### PR TITLE
Make var transforms work in bump

### PR DESCRIPTION
**Pull request by Elizafox**
My previous work has now come to a head and now var transforms work in bump with this patch.

Not all variables are available for substitution at the moment, as that would complicate the logic considerably. Variables that are presently supported for substitution:

- ${{package.name}}
- ${{package.version}}
- ${{package.epoch}}
- ${{build.arch}}

This is sufficient for now. All packages that use var transforms only transform the version at present.

### CODE REVIEW

File: pkg/config/vars.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	SubstitutionTargetsDestdir       = "${{targets.destdir}}"
+	SubstitutionSubPkgDir            = "${{targets.subpkgdir}}"
+	SubstitutionHostTripletGnu       = "${{host.triplet.gnu}}"
+	SubstitutionHostTripletRust      = "${{host.triplet.rust}}"
+	SubstitutionCrossTripletGnuGlibc = "${{cross.triplet.gnu.glibc}}"
+	SubstitutionCrossTripletGnuMusl  = "${{cross.triplet.gnu.musl}}"
+	SubstitutionBuildArch            = "${{build.arch}}"
+)
+
+func (cfg Configuration) GetVarsFromConfig() (map[string]string, error) {
```
Comment (kaniini) [inline pkg/config/vars.go:39]: Add a docstring?

File: pkg/config/vars.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+			return nil, err
+		}
+
+		nw[nk] = nv
+	}
+
+	return nw, nil
+}
+
+func (cfg Configuration) PerformVarSubstitutions(nw map[string]string) error {
```
Comment (kaniini) [inline pkg/config/vars.go:56]: Docstring?

File: pkg/util/map_subst.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+package util
+
+import (
+	"fmt"
+
+	"chainguard.dev/melange/pkg/cond"
+)
+
+func MutateStringFromMap(with map[string]string, input string) (string, error) {
```
Comment (kaniini) [inline pkg/util/map_subst.go:23]: Docstring?

Comment (Elizafox) [reply pkg/util/map_subst.go:23]: Original didn't have one. I fixed it.

File: pkg/config/vars.go (hunk 2 above)
Comment (Elizafox) [reply pkg/config/vars.go:56]: Added.

File: pkg/config/vars.go (hunk 1 above)
Comment (Elizafox) [reply pkg/config/vars.go:39]: Added.

**Review by kaniini — APPROVED** [review]

**Pull request merged**